examples/karpenter/main.tf

```diff
@@ -125,12 +125,44 @@ module "aws-eks-accelerator-for-terraform" {
     }
   }
 }
+# Creates Launch templates for Karpenter
+module "karpenter-launch-templates" {
```

Comment:
Where are these launch templates used? It seems that they are not referenced anywhere.

Reply:
Is it referenced in the custom resource `karpenter.sh/v1alpha5/Provisioner`? Do I have to deploy it manually?

Reply:
Yes, Check this example. `examples/karpenter/provisioners/default_provisioner_with_lanch_templates.yaml`

Reply:
Is it possible to auto-deploy the provisioner via terraform?

Reply:
You can have many types of provisioners and it depends upon the customers usecase. We feel that this should be managed outside the add-on deployment. 

One option is to use GitOps to maintain all the karpenter provisioners under one App

Reply:
I don't know how Karpenter works but I guess it won't do anything without a privisioner? In that case, I would suggest that the module should at least force the user to provide a provisioner through a mandatory variable and a hook in the module should deploy it. 

Looking at the default_provisioner.yaml it might be feasible to heuristically derive reasonable values from input variables/or outputs of other modules. But this is future work.

Reply:
Thanks @schwichti We will discuss this and find a rightway to handle this. I will raise a ticket for this